internal/pkg/instrumentation/bpf/net/http/client/bpf/probe.bpf.c

```diff
@@ -120,6 +146,49 @@ int uprobe_Transport_roundTrip(struct pt_regs *ctx) {
         bpf_printk("uprobe_Transport_roundTrip: Failed to get path from Request.URL");
     }
 
+    // get scheme from Request.URL
```

Comment:
Hi @damemi, based on our discussion from the SIG call today, it seems that the code is already using a BPF per-cpu array to store the httpRequest. Do you remember where did you get a stack limit error if the sizes of the arrays were made bigger?

Reply:
I wonder if the error might've been related to stack alignment?

Reply:
@grcevski I just tried reproducing it, but I had to add a lot more space to each field than I thought. I'll push my change so you can see the failure, but what it comes up with is this:

```
#0 8.736 bpf/probe.bpf.c:125:5: error: A call to built-in function 'memset' is not supported.
#0 8.736     __builtin_memset(httpReq, 0, sizeof(struct http_request_t));
#0 8.736     ^
#0 8.785 1 error generated.
#0 8.791 Error: can't execute clang: exit status 1
#0 8.792 exit status 1
#0 8.795 internal/pkg/instrumentation/bpf/net/http/client/probe.go:41: running "go": exit status 1
```

googling it I found github comments pointing to the 512 byte limit as causing that, but it could totally be something else I'm not aware of

Reply:
@grcevski you can see the failure [here](https://github.com/open-telemetry/opentelemetry-go-instrumentation/actions/runs/8991111452/job/24697949058?pr=790) from the commit I just pushed https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/790/commits/4a9ee82a8eb30a722bc7a484ec4a93334c567cc6

Reply:
Ah I think it's the use of `__buitin_memset` which can only clear up to 512 bytes. I think to workaround this problem we can avoid using memset and manually set each scalar field to 0 and each character array 0th byte to '\0'. So essentially, manually initialize all fields. It should be less operations too, since we'll be setting a lot less bytes to 0, e.g. maybe 30?

Perhaps this shouldn't hold up the PR, we can follow-up with another change if we wanted to increase the sizes.

Reply:
SGTM